api/jobset/v1alpha2/jobset_types.go

```diff
@@ -80,7 +80,6 @@ const (
 
 // JobSetSpec defines the desired state of JobSet
 // +kubebuilder:validation:XValidation:rule="!(has(self.startupPolicy) && self.startupPolicy.startupPolicyOrder == 'InOrder' && self.replicatedJobs.exists(x, has(x.dependsOn)))",message="StartupPolicy and DependsOn APIs are mutually exclusive"
-// +kubebuilder:validation:XValidation:rule="!(has(self.replicatedJobs[0].dependsOn))",message="DependsOn can't be set for the first ReplicatedJob"
```

Comment:
so this now allows users to set it on the first replicatedJob, what happens if they do?

Reply:
https://github.com/kubernetes-sigs/jobset/pull/819/files#diff-abd102e840ad3a94481819c05b8f6b1e8eda15c7b18483739bc9066460c9d2b8R162

I removed the CEL validation and moved it to an actual webhook.

I couldn't find any examples of using CEL to validate slice length so I gave up.